### ISSUE TITLE
custom_domain_verification_id output not being written to state file for azurerm_linux_function_app resources

### ISSUE THREAD

**Issue by shinji** [hashicorp/terraform-provider-azurerm#17444]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.4

### AzureRM Provider Version

3.11.0

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app

### Terraform Configuration Files

```hcl
provider "azurerm" {
  features {}
}

resource "azurerm_resource_group" "example" {
  name     = "example-resources"
  location = "West Europe"
}

resource "azurerm_storage_account" "example" {
  name                     = "linuxfunctionappsa"
  resource_group_name      = azurerm_resource_group.example.name
  location                 = azurerm_resource_group.example.location
  account_tier             = "Standard"
  account_replication_type = "LRS"
}

resource "azurerm_service_plan" "example" {
  name                = "example-app-service-plan"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location
  os_type             = "Linux"
  sku_name            = "Y1"
}

resource "azurerm_linux_function_app" "example" {
  name                = "example-linux-function-app"
  resource_group_name = azurerm_resource_group.example.name
  location            = azurerm_resource_group.example.location

  storage_account_name = azurerm_storage_account.example.name
  service_plan_id      = azurerm_service_plan.example.id

  site_config {}
}

output "custom-domain-verification-id" {
  value     = azurerm_linux_function_app.example.custom_domain_verification_id
  sensitive = true
}
```

### Debug Output/Panic Output

```shell
$ terraform output custom-domain-verification-id
""
```

### Expected Behaviour

```
$ terraform output custom-domain-verification-id
"CUSTOM DOMAIN VERIFICATION ID"
```

### Actual Behaviour

```
$ terraform output custom-domain-verification-id
""
```

### References

Similar bug: https://github.com/hashicorp/terraform-provider-azurerm/issues/17042

**Comment by reifnir** [comment]
I can confirm some more version combinations in which this is currently broken
- Terraform `1.2.4` and azurerm `3.9.0`
- Terraform `1.2.4` and azurerm `3.10.0`
- Terraform `1.2.4` and azurerm `3.11.0`
- Terraform `1.2.4` and azurerm `3.12.0`
- Terraform `1.2.3` and azurerm `3.12.0`
- Terraform `1.2.2` and azurerm `3.12.0`
- Terraform `1.2.1` and azurerm `3.12.0`
- Terraform `1.2.0` and azurerm `3.12.0`
- Terraform `1.3.0-alpha20220622` and azurerm `3.12.0`

**Comment by reifnir** [comment]
I made a PR [here](https://github.com/hashicorp/terraform-provider-azurerm/pull/17485) with the fix, but the pipeline is on hold until a maintainer gives it the 👍 (since I've never contributed before)

**Comment by tombuildsstuff** [comment]
Closing since this has been [fixed in v3.14.0](https://github.com/hashicorp/terraform-provider-azurerm/pull/17485) - thanks @reifnir
